tachyon/zk/base/polynomial_pointer.h

```diff
@@ -0,0 +1,36 @@
+#ifndef TACHYON_ZK_BASE_POLYNOMIAL_POINTER_H_
+#define TACHYON_ZK_BASE_POLYNOMIAL_POINTER_H_
+
+namespace tachyon::zk {
+
+template <typename PCSTy>
+class PolynomialPointer {
+ public:
+  using F = typename PCSTy::Field;
+  using Poly = typename PCSTy::Poly;
+
+  PolynomialPointer() = default;
+  PolynomialPointer(const Poly& poly, const F& blind)
+      : poly_(poly), blind_(blind) {}
+
+  const Poly& poly() const { return poly_; }
+
+  // Return |true| if |this| and |other| reference to same instance.
+  constexpr bool operator==(const PolynomialPointer& other) const {
+    return this == &other;
+  }
+
+  constexpr bool operator!=(const PolynomialPointer& other) const {
+    return !(this == &other);
+  }
+
+ private:
+  // Not owned.
+  const Poly& poly_;
+  // Not owned.
+  const F blind_;
```

Comment:
Do you intend this?

```suggestion
  const F& blind_;
```

Reply:
I thought copying the random number would prevent its reuse, but it seems that this will not help to prevent the reuse of randomness.
Therefore, I will modify the code to reference the 'blind' variable as well.